presto-main/src/main/java/io/prestosql/eventlistener/EventListenerManager.java

```diff
@@ -39,75 +41,97 @@
+        if (configFiles.isEmpty()) {
+            if (!CONFIG_FILE.exists()) {
+                return;
+            }
+            configFiles = ImmutableList.of(CONFIG_FILE);
         }
-
-        Map<String, String> properties = new HashMap<>(loadProperties(configFile));
-
-        String name = properties.remove(NAME_PROPERTY);
-        checkState(!isNullOrEmpty(name), "Access control configuration %s does not contain '%s'", configFile, NAME_PROPERTY);
-
-        setConfiguredEventListener(name, properties);
+        List<EventListener> eventListeners =
+                configFiles.stream().map(this::createEventListener).collect(Collectors.toList());
+        this.configuredEventListeners.set(eventListeners);
     }
 
-    @VisibleForTesting
-    protected void setConfiguredEventListener(String name, Map<String, String> properties)
+    private EventListener createEventListener(File configFile)
     {
-        requireNonNull(name, "name is null");
-        requireNonNull(properties, "properties is null");
-
-        log.info("-- Loading event listener --");
-
-        EventListenerFactory eventListenerFactory = eventListenerFactories.get(name);
-        checkState(eventListenerFactory != null, "Event listener '%s' is not registered", name);
-
-        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(eventListenerFactory.getClass().getClassLoader())) {
-            EventListener eventListener = eventListenerFactory.create(ImmutableMap.copyOf(properties));
-            this.configuredEventListener.set(Optional.of(eventListener));
+        log.info("-- Loading event listener %s --", configFile);
+        configFile = configFile.getAbsoluteFile();
+        Map<String, String> properties;
+        try {
+            properties = new HashMap<>(loadProperties(configFile));
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException("Failed to read configuration file: " + configFile, e);
         }
 
-        log.info("-- Loaded event listener %s --", name);
+        String name = properties.remove(EVENT_LISTENER_NAME_PROPERTY);
+        checkArgument(!isNullOrEmpty(name), "EventListener plugin configuration for %s does not contain %s", configFile,
+                EVENT_LISTENER_NAME_PROPERTY);
+        EventListenerFactory eventListenerFactory = eventListenerFactories.get(name);
+        EventListener eventListener = eventListenerFactory.create(properties);
+        log.info("-- Loaded event listener %s --", configFile);
+        return eventListener;
     }
 
     public void queryCompleted(QueryCompletedEvent queryCompletedEvent)
     {
-        if (configuredEventListener.get().isPresent()) {
-            configuredEventListener.get().get().queryCompleted(queryCompletedEvent);
+        for (EventListener listener : configuredEventListeners.get()) {
+            try {
+                listener.queryCompleted(queryCompletedEvent);
+            }
+            catch (Exception e) {
```

Comment:
This should be `RuntimeException` since the event listener doesn't throw any checked exceptions. However, I'm thinking we should make this `Throwable`. While we hope plugins don't throw `Error`, if they do, there's nothing we can do, so we might as well log it and continue.

Same for the other events.